source/fundamentals/connections/connection-options.txt

```diff
@@ -0,0 +1,201 @@
+       specified in the connection URI.
+
+For a full list of connection options, see the `ClientOptions API
+documentation
+<{+api+}/mongo/options#ClientOptions>`__.
+
+.. _golang-timeout-setting:
+
+Single Timeout Setting
+----------------------
+
+You can set a single ``Timeout`` option on your ``Client`` instance to
+specify the maximum amount of time that a single operation can take to
+execute.
+
+Set a timeout by calling the ``SetTimeout()`` method when specifying
+options for your ``Client`` instance or by specifying the
+``timeoutMS`` option in your connection URI. By default, all
+``Database``, ``Collection``, ``Session``, ``ChangeStream``, and
+``Bucket`` instances elsewhere in your application inherit the
+``Timeout`` option from ``Client`` if you do not set a different timeout
+on specific operations in the operation's Context.
+
+If you set a timeout on a Context passed into an operation, the driver uses
+that value for the operation. If you do not specify a Context timeout,
+the operation Context derives the timeout from the ``Client`` instance.
+
+.. note:: Retries under Timeout Specification
+
+   When using default settings, if you set a ``Timeout`` option on your ``Client``
+   and your operation requires a retry, the driver retries the operation
+   as many times as possible before the timeout expires. Once the timeout
+   expires, the driver returns a timeout error. Versions 1.1 and later
+   of the {+driver-short+} enable retryable reads and writes by default.
+   See the Server manual for more information about :ref:`retryable
+   reads <retryable-reads>` and :manual:`retryable writes </core/retryable-writes/>`.
+
+Timeout Examples
+~~~~~~~~~~~~~~~~
+
+This section provides examples that demonstrate different ways to set the
+``Timeout`` setting in your application.
+
+Client Option
+^^^^^^^^^^^^^
+
+The following code shows how to set the ``Timeout`` option on a ``Client``
+by using the ``SetTimeout()`` method:
+
+.. code-block:: go
+
+   opts := options.Client().SetTimeout(5 * time.Second)
+   client, err := mongo.Connect(opts)
+
+Connection String Option
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+The following example shows how to set a single timeout by using the
+``timeoutMS`` URI option. Then, the code executes an insert operation
+that inherits the timeout:
+
+.. code-block:: go
+   :emphasize-lines: 1, 5
+
+   uri := "mongodb://user:pass@sample.host:27017/?timeoutMS=5000"
+   client, err := mongo.Connect(options.Client().ApplyURI(uri))
+   
+   ...
+   coll.InsertOne(context.Background(), doc)
+
+Multiple Timeouts
+^^^^^^^^^^^^^^^^^
+
+The following example shows how to set the ``Timeout`` option on a
+``Client`` instance. Then, the code sets a different timeout within the
+Context passed to the ``InsertOne()`` method:
+
+.. code-block:: go
+   :emphasize-lines: 1, 5
+
+   opts := options.Client().SetTimeout(5 * time.Second)
+   client, err := mongo.Connect(opts)
```

Comment:
I don't think it's necessary to show this. For an operation-level timeout, it doesn't matter if a client-timeout is set or not, the operation timeout takes priority. Suggest removing this linea nd updating the description of this example:

```
The following example shows how to set an operation-level timeout, which 
takes priority over a client-level timeout if one is defined:
```